packages/components/src/custom-select/types.ts

```diff
@@ -0,0 +1,38 @@
+// TO DO
+export type CustomSelectProps = {
+	/**
+	 * The children elements. This should be composed
+	 * of CustomSelect.Item components.
+	 */
+	children: React.ReactNode;
+	/**
+	 * An optional default value for the control.
+	 * If left undefined, the first non-disabled
+	 * item will be used
+	 */
+	defaultValue?: any;
+	/**
+	 * Label for the control.
+	 */
+	label?: string;
+	/**
+	 * A function that receives the new value of the input.
+	 */
+	onChange?: Function;
+	/**
+	 * The size of the select.
+	 */
+	size?: 'default' | 'large';
+};
+
+export type CustomSelectItemProps = {
+	/**
+	 * The children elements to be displayed for the select options.
+	 */
+	children: React.ReactNode;
+	/**
+	 * The current value of the select. Can be used in lieu of
+	 * children if only a string is needed.
+	 */
+	value?: any;
```

Comment:
Currently, this could be a `ReactNode`, `string`, or `undefined` but I'm unsure of how to type it. The intention here is to avoid duplication, in scenarios where the child is an only child or string, it could then be used for the value without explicitly defining it.

Reply:
Resolved in https://github.com/WordPress/gutenberg/pull/55790/commits/56d9f7fb418e2a1c07032a4164cc6790784b7337